File: dexguard-sample/src/main/java/ivonhoe/dexguard/example/MainActivity.java
package ivonhoe.dexguard.example;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;

import ivonhoe.dexguard.anotaion.MethodGuard;

public class MainActivity extends AppCompatActivity {

    @Override
    @MethodGuard()
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        HelloWorld helloWorld = new HelloWorld();
        helloWorld.testDexGuard();
    }

    @Override
    @MethodGuard()
    protected void onResume() {
        super.onResume();
    }

    @Override
    @MethodGuard()
    protected void onStop() {
        super.onStop();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }

    private class HelloWorld {

        @MethodGuard()
        private void testDexGuard() {
            Log.d("HelloWorld", "test dexguard");
        }

        private void testDexGuard(int param) {
            Log.d("HelloWorld", "test dexguard");
        }

        private void testDexGuard(String param) {
            Log.d("HelloWorld", "test dexguard");
        }

        @MethodGuard()
        private void testDexGuard2() {
            Log.d("HelloWorld", "test dexguard2");
        }
    }
}

File: mine-example/src/main/java/ivonhoe/android/mine/example/MainActivity.kt
package ivonhoe.android.mine.example

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import ivonhoe.android.mine.trace.LongTimeTrace

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        testMethod1()
    }

    fun testMethod() {
        var startTime = LongTimeTrace.methodStart()

        // do test
        Thread.sleep(200);

        LongTimeTrace.methodEnd(true, startTime);
    }

    fun testMethod1() {
        var test1 = Test1();

        test1.testMethod1()
    }
}